controllers/runnerjob_controller.go

```diff
@@ -0,0 +1,160 @@
+//+kubebuilder:rbac:groups=actions.summerwind.dev,resources=runnerjobs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=batch,resources=jobs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=batch,resources=jobs/status,verbs=get
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the RunnerJob object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.6.4/pkg/reconcile
+func (r *RunnerJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("runnerjob", req.NamespacedName)
+
+	log.Info("reading the runner job", "RunnerJob", req.NamespacedName)
+	defer log.Info("finishing reconciling")
+
+	var runnerJob v1alpha1.RunnerJob
+	if err := r.Get(ctx, req.NamespacedName, &runnerJob); err != nil {
+		log.Info("resource could not be loaded", "error", err.Error())
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if !runnerJob.ObjectMeta.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	}
+
+	var childJobs batchv1.JobList
+	if err := r.List(ctx, &childJobs, client.InNamespace(req.Namespace), client.MatchingFields{jobOwnerKey: req.Name}); err != nil {
+		log.Error(err, "unable to list child Jobs")
+		return ctrl.Result{}, err
+	}
+
+	for _, childJob := range childJobs.Items {
+		log.Info("Child Job", "Status", childJob.Status)
+	}
+
+	switch len(childJobs.Items) {
+	case 0:
+		log.Info("Create job")
+		if err := r.createJob(ctx, runnerJob); err != nil {
+			log.Info("Failed to create a job from a RunnerJob, requeueing in 30s", runnerJob.Name, err.Error())
+			return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
+		}
+	case 1:
+		log.Info("We have 1 child job")
+		childJob := childJobs.Items[0]
+		// if runner pod is running, don't reconcile again on this state
+		if childJob.Status.Active > 0 {
+			return ctrl.Result{}, nil
+		}
+
+		if childJob.Status.Succeeded > 0 {
+			// cleanup runner job
+			if err := r.Delete(ctx, &runnerJob); err != nil {
+				log.Info("Failed to delete runner job", "RunnerJob", runnerJob.Name)
+				return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
+			}
+			return ctrl.Result{}, nil
+		}
+
+	default:
+		log.V(0).Info("Found runner job with more than one kubernetes job", "RunnerJob", runnerJob.Name)
+		return ctrl.Result{}, fmt.Errorf("Found runner job with more than one kubernetes job")
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *RunnerJobReconciler) createJob(ctx context.Context, runnerJob v1alpha1.RunnerJob) error {
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:      make(map[string]string),
```

Comment:
Maybe we should add a common label? If someone has multiple scale sets in a single k8s cluster then the label could help listing across namespaces and such.